stylesheets/_component.dropzone.scss

```diff
@@ -0,0 +1,319 @@
+/**
+ * Dropzone
+ */
+
+$dropzone-file-size: .5em;
+$dropzone-thumb-size: 4em;
+
+.dropzone {
+    border: 2px dashed color(gray, light);
+    border-radius: $border-radius;
+    color: color(text, base);
+    padding: 1.25em;
+    position: relative;
+    text-align: center;
+}
+
+// the element responsible for creating success and error flashes
+.dropzone__state {
+    bottom: 0;
+    content: '';
+    height: 100%;
+    left: 0;
+    position: absolute;
+    pointer-events: none;
+    right: 0;
+    top: 0;
+    width: 100%;
+    z-index: 0;
+}
+
+.dropzone__file-wrapper {
+    display: flex;
+    flex-wrap: wrap;
+    margin: (1em - $dropzone-file-size) -#{$dropzone-file-size} -#{$dropzone-file-size};
+}
+
+.dropzone__icon-wrapper {
+    padding: 1.75em .5em .5em;
+    position: relative;
+}
+
+/**
+ * Main Icon
+ */
+
+.dropzone__icon {
+    color: color(gray, dark);
+    font-size: 4em;
+    text-align: center;
+}
+
+.dropzone__icon--small {
+    font-size: 1em;
+    position: absolute;
+    will-change: transform;
+}
+
+$dropzone-icon-rotation: 30deg;
```

Comment:
I've moved the file size variables a little further down for consistency, I prefer having config variables in throwing distance from their context, rather than the put all variables at the top approach.